test/src/ui/splitpanel.spec.ts

```diff
@@ -0,0 +1,1191 @@
+  }
+
+  protected onChildAdded(msg: ChildMessage): void {
+    super.onChildAdded(msg);
+    this.methods.push('onChildAdded');
+  }
+
+  protected onChildRemoved(msg: ChildMessage): void {
+    super.onChildRemoved(msg);
+    this.methods.push('onChildRemoved');
+  }
+}
+
+
+class LogSplitLayout extends SplitLayout {
+
+  methods: string[] = [];
+
+  protected attachWidget(index: number, widget: Widget): void {
+    super.attachWidget(index, widget);
+    this.methods.push('attachWidget');
+  }
+
+  protected moveWidget(fromIndex: number, toIndex: number, widget: Widget): void {
+    super.moveWidget(fromIndex, toIndex, widget);
+    this.methods.push('moveWidget');
+  }
+
+  protected detachWidget(index: number, widget: Widget): void {
+    super.detachWidget(index, widget);
+    this.methods.push('detachWidget');
+  }
+
+  protected onLayoutChanged(msg: Message): void {
+    super.onLayoutChanged(msg);
+    this.methods.push('onLayoutChanged');
+  }
+
+  protected onAfterShow(msg: Message): void {
+    super.onAfterShow(msg);
+    this.methods.push('onAfterShow');
+  }
+
+  protected onAfterAttach(msg: Message): void {
+    super.onAfterAttach(msg);
+    this.methods.push('onAfterAttach');
+  }
+
+  protected onChildShown(msg: ChildMessage): void {
+    super.onChildShown(msg);
+    this.methods.push('onChildShown');
+  }
+
+  protected onChildHidden(msg: ChildMessage): void {
+    super.onChildHidden(msg);
+    this.methods.push('onChildHidden');
+  }
+
+  protected onResize(msg: ResizeMessage): void {
+    super.onResize(msg);
+    this.methods.push('onResize');
+  }
+
+  protected onUpdateRequest(msg: Message): void {
+    super.onUpdateRequest(msg);
+    this.methods.push('onUpdateRequest');
+  }
+
+  protected onFitRequest(msg: Message): void {
+    super.onFitRequest(msg);
+    this.methods.push('onFitRequest');
+  }
+}
+
+
+class LogWidget extends Widget {
+
+  methods: string[] = [];
+
+  protected onAfterAttach(msg: Message): void {
+    super.onAfterAttach(msg);
+    this.methods.push('onAfterAttach');
+  }
+
+  protected onBeforeDetach(msg: Message): void {
+    super.onBeforeDetach(msg);
+    this.methods.push('onBeforeDetach');
+  }
+
+  protected onAfterShow(msg: Message): void {
+    super.onAfterShow(msg);
+    this.methods.push('onAfterShow');
+  }
+
+  protected onUpdateRequest(msg: Message): void {
+    super.onUpdateRequest(msg);
+    this.methods.push('onUpdateRequest');
+  }
+}
+
+
+class HandleRenderer implements SplitLayout.IHandleRenderer {
+
+  createHandleNode(): HTMLElement {
+    let node = document.createElement('div');
+    node.className = 'p-SplitPanel-handle';
+    return node;
+  }
+}
+
+
+describe('ui/splitpanel', () => {
+
+  describe('SplitPanel', () => {
+
+    describe('.createLayout()', () => {
+
+      it('should create a split layout for a split panel', () => {
+        let layout = SplitPanel.createLayout();
+        expect(layout instanceof SplitLayout).to.be(true);
```

Comment:
Not that there's any sort of functional difference, but for these, I've been doing:

`expect(layout).to.be.a(SplitLayout);`